### ISSUE TITLE
Redirect Stream to other URL

### ISSUE THREAD

**Issue by nicedevil007** [foosel/OctoPrint#3465]
Hey all out there :)

I like the included option to stream a video of my prints but I have a little problem with it.
I'm using a Sophos UTM Firewall in my home network and use it as reverseproxy as well.

Unfurtunately I'm not able to autoredirect the incoming traffic on my dns like
webcam.nicedevil.io to webcam.nicedevil.io/webcam/?action=stream.

That is because of the questionmark in this type of URL.

Now I just want to ask if it is possible right now to change the used URL for the stream or add this as a new feature to octoprint?

Thank you all in advance.

**Comment by GitIssueBot** [comment]
Hi @nicedevil007,

It looks like there is some **information missing** from your bug report that will be needed in order to solve the problem. Read the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md) which will provide you with a template to fill out here so that your bug report is ready to be investigated (I promise I'll go away then too!).

If you did not intend to report a bug but wanted to **request a feature or brain storm** about some kind of development, please take special note of the title format to use as described in the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md).

**Please do not abuse the bug tracker as a support forum** - that can be found at [discourse.octoprint.org](https://discourse.octoprint.org). Go there for any kind of issues with network connectivity, webcam functionality, printer detection or any other kind of such support requests or general questions.

Also **make sure you are at the right place** - this is the bug tracker of the official version of OctoPrint, not the Raspberry Pi image OctoPi nor any unbundled third party OctoPrint plugins or unofficial versions. Make sure too that you have **read through the [Frequently Asked Questions](http://faq.octoprint.org)** and searched the [**existing tickets**](https://github.com/foosel/OctoPrint/search?q=&ref=cmdform&type=Issues) for your problem - try multiple search terms please.

I'm marking this one now as needing some more information. Please understand that if you do not provide that information within the next two weeks (until 2020-03-12 22:30 UTC) I'll close this ticket so it doesn't clutter the bug tracker. This is nothing personal, so please just be considerate and help the maintainers solve this problem quickly by following the guidelines linked above. Remember, the less time the devs have to spend running after information on tickets, the more time they have to actually solve problems and add awesome new features. Thank you!

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being, so don't expect any replies from me :) Your ticket is read by humans too, I'm just not one of them.

**Comment by foosel** [comment]
The webcam server is not part of OctoPrint but rather mjpg-streamer which comes bundled with OctoPi (see https://community.octoprint.org/t/what-is-the-difference-between-octoprint-and-octopi-are-they-the-same-thing/185). 

Your request also sounds like something better suited for [the community forum](https://community.octoprint.org) rather than the issue tracker of either project.

**Comment by nicedevil007** [comment]
I found a solution with a friends help now. Just in case anyone is trying the same to achieve.

Change the settings of OctoPi's haproxy (/etc/haproxy/haproxy.cfg)

```
backend webcam
        acl p_ext_jpg path_end -i ?action=stream
        reqrep ^([^\ :]*)\ /webcam/(.*)     \1\ /\2
        server webcam1  127.0.0.1:8080
        http-request set-path /?action=stream if !p_ext_jpg
        errorfile 503 /etc/haproxy/errors/503-no-webcam.http
```

1st and 4th of this config is what we need.
Then if someone is going on https://urltoOctoPrint/webcam it will redirect to https://urltoOctoPrint/webcam/?action=stream

In my case I was able to permanently redirect all traffice to the main URL to /webcam and then it showing the stream.
